src/lib/components/molecules/canvas-map/lib/layers/VectorLayer.js

```diff
@@ -3,8 +3,35 @@ import { Style, Stroke } from "../styles"
 import { combineExtents } from "../util/extent"
 import { Dispatcher, MapEvent } from "../events"
 import { VectorSource } from "../sources/VectorSource"
+import { useEffect, useContext } from "preact/hooks"
+import { MapContext } from "$molecules/canvas-map/context/MapContext"
+
+/** @typedef {Omit<ConstructorParameters<typeof VectorLayer>[0], "source">} VectorLayerOptions */
+/** @typedef {VectorLayerOptions & { features: import("../Feature").Feature[] }} VectorLayerComponentProps */
 
 export class VectorLayer {
+  /** @param {VectorLayerComponentProps} props */
+  static Component({ features, ...options }) {
+    const { map } = useContext(MapContext)
+
+    useEffect(() => {
+      if (!map) return
+
+      const controller = VectorLayer.with(features, options)
+      map.addLayer(controller)
```

Comment:
I think there is an edge case here that needs to be taken into account. If for whatever reason you want to hide the background layer and then unhide it later, it should still be rendered as the first layer.

Perhaps we can solve this in `MapContext.js` by doing something like this?
```js
useEffect(() => {
  const layers = toChildArray(children)
  const shouldUpdateMapLayers = arrayEquals(map.layers, children)
  // ...update map layers
}, [children])
```